amplitude.js

```diff
@@ -270,15 +252,135 @@ Amplitude.prototype.runQueuedFunctions = function () {
   this._q = []; // clear function queue after running
 };
 
-Amplitude.prototype._loadSavedUnsentEvents = function(unsentKey, queue) {
-  var savedUnsentEventsString = localStorage.getItem(unsentKey);
+/**
+ * Set an item in local storage. Decorate item key with the api key
+ * to avoid collisions.
+ *
+ * @property item: name of the item to store
+ * @property value: the value to store
+ */
+Amplitude.prototype.setInStorage = function(item, value) {
+  var key = item + '_' + this.options.apiKey.slice(0, 6);
+  if (value) {
+    this.storage.setItem(key, value);
+    return;
+  }
+  // if value is null, then remove from storage
```

Comment:
In this implementation we treat them separately. Both are allowed in the SDK (for userId and deviceId) in that they will just get filtered out by the server during ingestion. You might say well if it will get filtered out, then we should just not store empty strings to save space in local storage. I feel if a user explicitly sets something to empty string, they might get confused when they check the value and see that we've turned it into null for them.